src/ctap/command.rs

```diff
@@ -132,23 +137,19 @@ impl TryFrom<cbor::Value> for AuthenticatorMakeCredentialParameters {
         )?)?;
 
         let cred_param_vec = read_array(ok_or_missing(param_map.get(&cbor_unsigned!(4)))?)?;
-        let mut pub_key_cred_params = vec![];
-        for cred_param_map_value in cred_param_vec {
-            let cred_param_map = read_map(cred_param_map_value)?;
-            let cred_type = PublicKeyCredentialType::try_from(ok_or_missing(
-                cred_param_map.get(&cbor_text!("type")),
-            )?)?;
-            let alg = read_integer(ok_or_missing(cred_param_map.get(&cbor_text!("alg")))?)?;
-            pub_key_cred_params.push((cred_type, alg));
-        }
+        let pub_key_cred_params = cred_param_vec
+            .iter()
+            .map(PublicKeyCredentialParameter::try_from)
+            .collect::<Result<Vec<PublicKeyCredentialParameter>, Ctap2StatusCode>>()?;
```

Comment:
I'm surprised collect returns a result of vec, rather than a vec of results.

Reply:
This is because `Result<V, E>` implements `FromIterator<Result<A, E>>` when `V` implements `FromIterator<A>`. This is also documented in https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect:

> A collection of Result<T, E> can be thought of as single Result<Collection<T>, E>.

Reply:
Nice!